Stratis.Bitcoin/Features/MemoryPool/MempoolBehavior.cs

```diff
@@ -131,6 +263,73 @@ private Task AttachedNode_MessageReceivedAsync(Node node, IncomingMessage messag
             return Task.CompletedTask;
         }
 
+        #endregion
+
+        #region Message Processing
+
+        /// <summary>
+        /// Send the memory pool payload to the attached node.
+        /// Gets the transaction info from the memory pool and sends to the attached node.
+        /// </summary>
+        /// <param name="node">Node Sending the message</param>
+        /// <param name="message">The payload</param>
+        /// <returns>The asynchronous task.</returns>
+        private async Task SendMempoolPayload(Node node, MempoolPayload message)
+        {
+            Guard.Assert(node == this.AttachedNode); // just in case
+
+            if (!this.CanSend)
+                return;
+
+            //if (!(pfrom->GetLocalServices() & NODE_BLOOM) && !pfrom->fWhitelisted)
+            //{
+            //	LogPrint("net", "mempool request with bloom filters disabled, disconnect peer=%d\n", pfrom->GetId());
+            //	pfrom->fDisconnect = true;
+            //	return true;
+            //}
+
+            //if (connman.OutboundTargetReached(false) && !pfrom->fWhitelisted)
+            //{
+            //	LogPrint("net", "mempool request with bandwidth limit reached, disconnect peer=%d\n", pfrom->GetId());
+            //	pfrom->fDisconnect = true;
+            //	return true;
+            //}
+
+            var vtxinfo = await this.manager.InfoAllAsync();
+            var filterrate = Money.Zero;
+
+            // TODO: implement minFeeFilter
+            //{
+            //	LOCK(pto->cs_feeFilter);
+            //	filterrate = pto->minFeeFilter;
+            //}
+
+            var sends = await this.manager.MempoolScheduler.WriteAsync(() =>
+            {
+                var ret = new List<TxMempoolInfo>();
+                foreach (var txinfo in vtxinfo)
+                {
+                    var hash = txinfo.Trx.GetHash();
+                    this.inventoryTxToSend.Remove(hash);
+                    if (filterrate != Money.Zero)
+                        if (txinfo.FeeRate.FeePerK < filterrate)
+                            continue;
+                    this.filterInventoryKnown.TryAdd(hash, hash);
+                }
+                return ret;
+            });
+
+            await this.SendAsTxInventory(node, sends.Select(s => s.Trx.GetHash()));
+            this.LastMempoolReq = this.manager.DateTimeProvider.GetTime();
+        }
+
+        /// <summary>
+        /// Asynchronous processing of inventory payload from the node.
```

Comment:
The word "Asynchronous" can be omitted here as the method is `async` and its name ends with `Async`.